bugsnag-android-core/src/main/java/com/bugsnag/android/ThreadState.kt

```diff
@@ -109,16 +113,24 @@ internal class ThreadState @Suppress("LongParameterList") constructor(
 
         // Keep the lowest ID threads (ordered). Anything after maxThreadCount is lost.
         // Note: We must ensure that currentThread is always present in the final list regardless.
+
+        val reportThreads = mutableListOf<Thread>()
+        val timeout = SystemClock.elapsedRealtime() + maxTime
         val keepThreads = allThreads.sortedBy { it.id }.take(maxThreadCount)
+        keepThreads.forEach {
+            if (SystemClock.elapsedRealtime() < timeout) {
+                reportThreads.add(toBugsnagThread(it))
+            }
+        }
```

Comment:
this might be better expressed as a `for` loop with a `break`?

```suggestion
        for(thread in keepThreads) {
            if (SystemClock.elapsedRealtime() >= timeout) {
                break
            }
            reportThreads.add(toBugsnagThread(it))
        }
```